src/PhpTemplateRenderer.php

```diff
@@ -4,21 +4,29 @@
 
 namespace Yiisoft\View;
 
+use Yiisoft\View\Exception\ContentCantBeFetched;
+
 class PhpTemplateRenderer implements TemplateRendererInterface
 {
     public function render(View $view, string $template, array $params): string
     {
-        $renderer = function () {
-            extract(func_get_arg(1), EXTR_OVERWRITE);
-            require func_get_arg(0);
+        $renderer = function (string $template, array $params) {
+            extract($params, EXTR_OVERWRITE);
+            require $template;
         };
 
         $obInitialLevel = ob_get_level();
         ob_start();
         PHP_VERSION_ID >= 80000 ? ob_implicit_flush(false) : ob_implicit_flush(0);
         try {
             $renderer->bindTo($view)($template, $params);
-            return ob_get_clean();
+
+            $content = ob_get_clean();
+            if (is_string($content)) {
+                return $content;
+            } else {
+                throw new ContentCantBeFetched();
```

Comment:
That won't be called ever.

Reply:
It will be when
`
class SiteController
{
    public function index(WebView $view, DataResponseFactoryInterface $responseFactory): ResponseInterface
    {
        return $responseFactory->createResponse($view->renderFile('/app/views/layout/main.php', [
            'content' => ''
        ]));
    }
}
`

`
declare(strict_types=1);

/**
 * @var \Yiisoft\View\WebView $this
 * @var string $content
 */
?>

<!DOCTYPE html>
<html lang="">
<head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <title>Yii Demo</title>
</head>

<body>

<?php
$this->beginBody();
?>

<main role="main" class="container">
    <?= $content; ?>
</main>

<?php
$this->endBody();
?>

</body>
</html>

<?php
$this->endPage(true);
`

In given example of view file  `$this->beginPage();`  is  missing